server/src/main/java/ch/bifrost/server/impl/keyexchange/DatagramSigningConverter.java

```diff
@@ -0,0 +1,94 @@
+package ch.bifrost.server.impl.keyexchange;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.security.cert.CertificateEncodingException;
+
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.NoSuchAlgorithmException;
+import java.security.InvalidKeyException;
+
+import org.bouncycastle.cert.jcajce.JcaX509CertificateHolder;
+import org.bouncycastle.asn1.ASN1Encodable;
+import org.bouncycastle.asn1.DERSequence;
+import org.bouncycastle.asn1.DERBitString;
+
+import com.google.common.base.Optional;
+
+import ch.bifrost.core.api.datagram.CounterpartAddress;
+import ch.bifrost.core.api.datagram.DatagramEndpoint;
+import ch.bifrost.core.api.datagram.DatagramMessage;
+import ch.bifrost.core.impl.datagram.InvalidDatagramException;
+
+public class DatagramSigningConverter implements DatagramEndpoint {
+
+	private DatagramEndpoint endpoint;
+	private Signature signature;
+	private ASN1Encodable certChain;
+	
+	@Override
+	public void close() throws IOException {
+		endpoint.close();
+	}
+
+	@Override
+	public void send(DatagramMessage packet)
+			throws IOException {
+		try {
+			ASN1Encodable asn1payload = new DERBitString(packet.getPayload(), 0);
```

Comment:
Wenn es statt ASN1Encodable keine Typesafe Klasse gibt, würde ich das in eine Extra-Klasse wrappen, damit der encoding und der decoding Teil in der gleichen Klasse ist (und nicht wie im Moment verteil auf diese Klasse und den DatagramVerifyingConverter).